tests/async.rs

```diff
@@ -210,13 +222,16 @@ fn error_response() {
         move |_, w| w.error(err.0, err.1.as_bytes()),
         |_| unreachable!(),
         |_, _, _| unreachable!(),
-    ).test(|db| {
-        db.query("SELECT a, b FROM foo").then(|r| {
+    ).test(move |db| {
+        db.query("SELECT a, b FROM foo").then(move |r| {
```

Comment:
Are these `move`s necessary? I'm not sure I immediately see why, though I could be missing something?

Reply:
I added them at rustc's suggestion, but I am not sure exactly when it decided those were necessary. I believe the problem came in once I added the 'static lifetime bounds for F & C of adding the additional bounds for F & C of the test functions. Without them it would give errors like:

```
error[E0373]: closure may outlive the current function, but it borrows `err`, which is owned by the current function
   --> tests/async.rs:222:9
    |
222 |         |_, w| w.error(err.0, err.1.as_bytes()),
    |         ^^^^^^         --- `err` is borrowed here
    |         |
    |         may outlive borrowed value `err`
    |
note: function requires argument type to outlive `'static`
   --> tests/async.rs:221:5
    |
221 | /     TestingShim::new(
222 | |         |_, w| w.error(err.0, err.1.as_bytes()),
223 | |         |_| unreachable!(),
224 | |         |_, _, _| unreachable!(),
225 | |     ).test(|db| {
    | |_____^
help: to force the closure to take ownership of `err` (and any other referenced variables), use the `move` keyword
    |
222 |         move |_, w| w.error(err.0, err.1.as_bytes()),
    |         ^^^^^^^^^^^

error[E0373]: closure may outlive the current function, but it borrows `err`, which is owned by the current function
   --> tests/async.rs:225:12
    |
225 |     ).test(|db| {
    |            ^^^^ may outlive borrowed value `err`
...
238 |                         &String::from_utf8(err.0.sqlstate().to_vec()).unwrap()
    |                                            --- `err` is borrowed here
    |
note: function requires argument type to outlive `'static`
   --> tests/async.rs:221:5
    |
221 | /     TestingShim::new(
222 | |         |_, w| w.error(err.0, err.1.as_bytes()),
223 | |         |_| unreachable!(),
224 | |         |_, _, _| unreachable!(),
...   |
249 | |         })
250 | |     })
    | |______^
help: to force the closure to take ownership of `err` (and any other referenced variables), use the `move` keyword
    |
225 |     ).test(move |db| {
    |            ^^^^^^^^^

```

Reply:
I think they should no longer be necessary now that you've switched to the current thread runtime, but could be wrong?

Reply:
I didn't follow up on my last comment before resolving, sorry. You are correct, they are no longer needed once switched to the current thread runtime. I thought, I had changed both cases but missed one, I will get this corrected with my next commit.